### PR TITLE
Refac/cas 1267 thread controller in offline plugin

### PR DESCRIPTION
**Pull request by ogkuzmin**
### 🎯 Goal

Refactor `ThreadController` and move logic to `OfflinePlugin`

### 🛠 Implementation details

1. Move data to `ThreadMutableState`
2. Move logic to `ThreadLogic`
3. Wrap side effects in listeners and invoke them when `ChatClient::getReplies` is invoked
4. Fix tests

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 1]
```diff
@@ -825,18 +825,39 @@ public class ChatClient internal constructor(
     }
 
     @CheckResult
-    public fun getReplies(messageId: String, limit: Int): Call<List<Message>> {
-        return api.getReplies(messageId, limit)
-    }
+    @InternalStreamChatApi
+    @ExperimentalStreamChatApi
+    public fun getRepliesInternal(messageId: String, limit: Int): Call<List<Message>> = api.getReplies(messageId, limit)
```
Comment (ogkuzmin) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:830]: Copy methods without side effects for bakcward compatibility when use `ThreadController`. Will be remove when `ChatDomain` and its' controllers are deprecated and hidden from public API

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt [hunk 2]
```diff
@@ -1035,24 +1040,6 @@ public class ChannelController internal constructor(
         return channel
     }
 
-    internal suspend fun loadOlderThreadMessages(
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt:1038]: Moved it to ThreadController directly

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/usecase/ThreadLoadMore.kt [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-internal class ThreadLoadMore(private val domainImpl: ChatDomainImpl) {
-    /**
-     * Loads more messages for the specified thread.
-     *
-     * @param cid The full channel id i. e. messaging:123.
-     * @param parentId The parentId of the thread.
-     * @param messageLimit How many new messages to load.
-     */
-    @CheckResult
-    operator fun invoke(cid: String, parentId: String, messageLimit: Int): Call<List<Message>> {
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/usecase/ThreadLoadMore.kt:19]: Moved this code directly to `ChatDomain`

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/usecase/GetThread.kt [hunk 4]
```diff
@@ -1,32 +0,0 @@
-package io.getstream.chat.android.offline.usecase
-
-import androidx.annotation.CheckResult
-import io.getstream.chat.android.client.call.Call
-import io.getstream.chat.android.client.call.CoroutineCall
-import io.getstream.chat.android.client.utils.Result
-import io.getstream.chat.android.offline.ChatDomainImpl
-import io.getstream.chat.android.offline.thread.ThreadController
-import io.getstream.chat.android.offline.utils.validateCid
-
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/usecase/GetThread.kt:10]: Moved this code directly to `ChatDomain`

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/usecase/GetThreadTest.kt [hunk 5]
```diff
@@ -1,62 +0,0 @@
...
-import kotlinx.coroutines.runBlocking
-import org.amshove.kluent.`should throw`
-import org.amshove.kluent.`with message`
-import org.amshove.kluent.invoking
-import org.junit.Before
-import org.junit.Test
-import org.junit.runner.RunWith
-
-@RunWith(AndroidJUnit4::class)
-internal class GetThreadTest {
```
Comment (ogkuzmin) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/usecase/GetThreadTest.kt:23]: Not actual anymore. Tests weren't the best. They only check `validateCid` function

**Review by amitkma — APPROVED** [review]
LGTM

**Pull request merged**